### PR TITLE
Retrieve volume types from cinder and use to make storageclasses

### PR DESCRIPTION
**Pull request by heytrav**
Query cinder to find volume types and return set of storageclass dictionaries.

### CODE REVIEW

**Review by JohnGarbutt — COMMENTED** [review]
This looks good, but I am curious if @mkjpryor is happy.

**Comment by mkjpryor** [comment]
This looks really great. My only question would be whether we might want to use template labels to override some of the options that currently come exclusively from global config?

**Comment by heytrav** [comment]
> This looks really great. My only question would be whether we might want to use template labels to override some of the options that currently come exclusively from global config?
> 
> @JohnGarbutt what do you think?

I altered it so most of the options can be overridden as labels.

File: magnum_capi_helm/driver.py [hunk 1]
```diff
@@ -610,6 +688,12 @@ def _update_helm_release(self, context, cluster, nodegroups=None):
                 if ng.role != NODE_GROUP_ROLE_CONTROLLER
             ],
             "addons": {
+                "openstack": {
+                    "csiCinder": {
+                        "defaultStorageClass": default_storageclass,
+                        "additionalStorageClasses": add_storageclasses,
```
Comment (mkjpryor) [inline magnum_capi_helm/driver.py:694]: Instead of returning a tuple, why not have `_storageclass_definitions` return this object? Seems cleaner.

**Review by mkjpryor — CHANGES_REQUESTED** [review]

File: magnum_capi_helm/driver.py (hunk 1 above)
Comment (heytrav) [reply magnum_capi_helm/driver.py]: Yes that makes sense. Changed.

**Review by mkjpryor — APPROVED** [review]
LGTM

**Comment by heytrav** [comment]
curious that the linter is failing on files that I didn't even edit.

**Comment by heytrav** [comment]
Hi  @mkjpryor, I might need some help understanding what is wrong with linting here. After rebasing on the previous commit it is failing on 3 files (2 of which I haven't touched). 

If I run black locally it reports all good:

```
❯ black $PWD --check -l 79
All done! ✨ 🍰 ✨
22 files would be left unchanged.
```

**Comment by heytrav** [comment]
> Hi @mkjpryor, I might need some help understanding what is wrong with linting here. After rebasing on the previous commit it is failing on 3 files (2 of which I haven't touched).
> 
> If I run black locally it reports all good:
> 
> ```
> ❯ black $PWD --check -l 79
> All done! ✨ 🍰 ✨
> 22 files would be left unchanged.
> ```

Disregard. Updating `black` to more recent version fixed the problem.

**Review by JohnGarbutt — APPROVED** [review]
Lets get this merged, looks good.

Sorry about the linting, I just pushed a fix to pin the version of black to stop that happening again.

**Review by JohnGarbutt — APPROVED** [review]
I like it, thank you!

**Pull request merged**